### PR TITLE
Add Standard Message Recipe schema

### PR DESCRIPTION
**Pull request by k88hudson**
Added a general schema for all messages in messaging system (the CFR one is not complete)

### CODE REVIEW

File: types/messaging.ts [hunk 1]
```diff
@@ -1,4 +1,54 @@
-export interface CfrMessage {
-  slug: string;
-  trigger: object;
+// TODO: This needs expanding
+interface MessageTrigger {
```
Comment (k88hudson) [inline types/messaging.ts:2]: Andrei just did a bunch of work in https://phabricator.services.mozilla.com/D76475 for trigger schemas

File: types/messaging.ts [hunk 2]
```diff
@@ -1,4 +1,54 @@
...
+
+  /** Identifier that routes the message definition to the right UI. */
+  template: string;
+
+  /** A JEXL expression representing targeting information */
+  targeting?: string;
+
+  trigger?: MessageTrigger;
+
+  /** If more than one message is valid for a, the message with the higher priority will be shown  */
```
Comment (pdehaan) [inline types/messaging.ts:21]: Is this missing a word? “valid for a, “

File: types/messaging.ts [hunk 3]
```diff
@@ -1,4 +1,54 @@
...
+  /**
+   * For messages types that are randomly chosen (e.g. snippets), determines the relative probabily
+   * of a message being shown. For example, a message with a weight of 200 will show up twice as
+   * often as other messages (since the default is 100).
+   *
+   * @default 100
+   */
+  weight?: number;
+
+  /* Definition of maximum impressions in a particular period, ogit tar groups of periods */
```
Comment (pdehaan) [inline types/messaging.ts:33]: ogit?

File: types/messaging.ts [hunk 4]
```diff
@@ -1,4 +1,54 @@
...
+  frequency?: {
```
Comment (mythmon) [inline types/messaging.ts:34]: Is the idea here that you would pass either `lifetime` or `custom`, but not both? Or can they be used in tandem?

Comment (k88hudson) [reply types/messaging.ts:34]: They can used in tandem, yes – for example, you could have a lifetime cap of 3 (i.e. the message would never be shown again after 3 impressions) but also a daily cap of 1

**Comment by mythmon** [comment]
Thanks!

**Pull request merged**